### PR TITLE
Move to NC-Internal Db-Types

### LINKED ISSUE #737: Minversion 21 - Move to NC internal db column Types
Hi there,

I found that your app uses some classes of Nextcloud's third party libraries. Since an upgrade of those for php8 compatibility, your app will have to be adjusted as well. See https://docs.nextcloud.com/server/latest/developer_manual/app_publishing_maintenance/upgrade-guide.html#doctrine-dbal for details. Your app uses `Doctrine\DBAL\Types\Type` which lost the old constants: https://github.com/doctrine/dbal/blob/3.0.x/UPGRADE.md#deprecated-type-constants.

Cheers

### PR DESCRIPTION
**Pull request by jotoeri**
Fixes #737 and thus #818  :)

### CODE REVIEW

**Comment by jotoeri** [comment]
Now also added the String-Length fix here. 🙃 - Fix #1152

File: lib/Migration/Version030000Date20220402100057.php [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+	 * @return null|ISchemaWrapper
+	 */
+	public function changeSchema(IOutput $output, Closure $schemaClosure, array $options): ?ISchemaWrapper {
+		/** @var ISchemaWrapper $schema */
+		$schema = $schemaClosure();
+
+		$update_necesssary = false;
+
+		// Change Type of Description from string to text. Necessary due to length restrictions.
+		if ($schema->hasTable('forms_v2_forms') && $schema->getTable('forms_v2_forms')->hasColumn('description')) {
```
Comment (nickvergessen) [inline lib/Migration/Version030000Date20220402100057.php:50]: This should be the case, so you should not need to check for it. Quite the opposite? When someone updates and this column/table is missing you want it to error so they know something is wrong.

Comment (jotoeri) [reply lib/Migration/Version030000Date20220402100057.php]: Interesting point, yes. We had until now always these hasTable() used.... 🤔

I changed it. 👍

File: appinfo/info.xml [hunk 2]
```diff
@@ -45,7 +45,7 @@
     <screenshot>https://raw.githubusercontent.com/nextcloud/forms/master/screenshots/forms3.png</screenshot>
 
     <dependencies>
-        <nextcloud min-version="20" max-version="24" />
+        <nextcloud min-version="21" max-version="24" />
```
Comment (nickvergessen) [inline appinfo/info.xml:48]: 21 is EOL already, could directly jump to 22

Comment (jotoeri) [reply appinfo/info.xml:48]: Jep, but that is in #1150, as it 'needs' one more adaption... ;)

**Review by skjnldsv — APPROVED** [review]

**Comment by jotoeri** [comment]
Fixing php8 separately.

**Pull request merged**